src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs

```diff
@@ -310,14 +315,14 @@ private void CreateTestExecutionContext(ITestListener listener)
             Context = new TestExecutionContext();
 
             // Apply package settings to the context
-            if (Settings.ContainsKey(FrameworkPackageSettings.DefaultTimeout))
-                Context.TestCaseTimeout = (int)Settings[FrameworkPackageSettings.DefaultTimeout];
-            if (Settings.ContainsKey(FrameworkPackageSettings.DefaultCulture))
-                Context.CurrentCulture = new CultureInfo((string)Settings[FrameworkPackageSettings.DefaultCulture], false);
-            if (Settings.ContainsKey(FrameworkPackageSettings.DefaultUICulture))
-                Context.CurrentUICulture = new CultureInfo((string)Settings[FrameworkPackageSettings.DefaultUICulture], false);
-            if (Settings.ContainsKey(FrameworkPackageSettings.StopOnError))
-                Context.StopOnError = (bool)Settings[FrameworkPackageSettings.StopOnError];
+            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultTimeout, out var timeout))
+                Context.TestCaseTimeout = (int)timeout;
+            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultCulture, out var culture))
+                Context.CurrentCulture = new((string)culture, false);
+            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultUICulture, out var uiCulture))
+                Context.CurrentUICulture = new((string)uiCulture, false);
```

Comment:
I don't think we use these inferred constructors anywhere else (at least at the moement), and I find the code harder to read when there is no type on either side of the assignment, so I think we should keep the old constructors (at least for now).
```suggestion
                Context.CurrentCulture = new CultureInfo((string)culture, false);
            if (Settings.TryGetValue(FrameworkPackageSettings.DefaultUICulture, out var uiCulture))
                Context.CurrentUICulture = new CultureInfo((string)uiCulture, false);
```